sync/src/test/java/tech/pegasys/teku/sync/AttestationManagerTest.java

```diff
@@ -25,45 +30,41 @@
 
 import com.google.common.eventbus.EventBus;
 import com.google.common.primitives.UnsignedLong;
-import java.util.List;
 import org.apache.tuweni.bytes.Bytes32;
 import org.junit.jupiter.api.AfterEach;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
 import tech.pegasys.teku.bls.BLSSignature;
 import tech.pegasys.teku.core.results.AttestationProcessingResult;
 import tech.pegasys.teku.datastructures.blocks.SignedBeaconBlock;
-import tech.pegasys.teku.datastructures.operations.AggregateAndProof;
+import tech.pegasys.teku.datastructures.forkchoice.DelayableAttestation;
 import tech.pegasys.teku.datastructures.operations.Attestation;
 import tech.pegasys.teku.datastructures.operations.AttestationData;
+import tech.pegasys.teku.datastructures.operations.IndexedAttestation;
 import tech.pegasys.teku.datastructures.operations.SignedAggregateAndProof;
 import tech.pegasys.teku.datastructures.state.Checkpoint;
 import tech.pegasys.teku.datastructures.util.DataStructureUtil;
 import tech.pegasys.teku.ssz.SSZTypes.Bitlist;
+import tech.pegasys.teku.statetransition.attestation.AggregatingAttestationPool;
 import tech.pegasys.teku.statetransition.attestation.ForkChoiceAttestationProcessor;
-import tech.pegasys.teku.statetransition.events.attestation.ProcessedAggregateEvent;
-import tech.pegasys.teku.statetransition.events.attestation.ProcessedAttestationEvent;
 import tech.pegasys.teku.statetransition.events.block.ImportedBlockEvent;
-import tech.pegasys.teku.util.EventSink;
 
 class AttestationManagerTest {
   private final DataStructureUtil dataStructureUtil = new DataStructureUtil();
   private final EventBus eventBus = new EventBus();
-  private final PendingPool<DelayableAttestation> pendingAttestations =
-      PendingPool.createForAttestations();
-  private final FutureItems<DelayableAttestation> futureAttestations =
-      new FutureItems<>(DelayableAttestation::getEarliestSlotForForkChoiceProcessing);
 
-  private final ForkChoiceAttestationProcessor attestationProcessor =
+  private AggregatingAttestationPool attestationPool = mock(AggregatingAttestationPool.class);
+  private ForkChoiceAttestationProcessor attestationProcessor =
       mock(ForkChoiceAttestationProcessor.class);
-  private final List<ProcessedAttestationEvent> processedAttestationEvents =
-      EventSink.capture(eventBus, ProcessedAttestationEvent.class);
-  private final List<ProcessedAggregateEvent> processedAggregateEvents =
-      EventSink.capture(eventBus, ProcessedAggregateEvent.class);
+  private final PendingPool<DelayableAttestation> pendingAttestations =
+      spy(PendingPool.createForAttestations());
```

Comment:
ewww, spies...  All the checks on on these can be done quite effectively just using it's `size` method to see if anything has been added or not.

I'd also suggest adding a method:
```
private void verifyAttestationProcessed(final Attestation attestation) {
    ArgumentCaptor<DelayableAttestation> captor =
        ArgumentCaptor.forClass(DelayableAttestation.class);
    verify(attestationProcessor).processAttestation(captor.capture());
    assertThat(captor.getValue().getAttestation()).isSameAs(attestation);
  }
```

which will let you be more specific instead of using `verify(attestationProcessor).processAttestation(attestation)`.

Reply:
Done :D. Thanks for the suggestion.